OptimizelySDK/Optimizely.cs

```diff
@@ -666,22 +668,19 @@ public string GetFeatureVariableString(string featureKey, string variableKey, st
         {
             if (experiment.IsExperimentRunning)
             {
-                var impressionEvent = EventBuilder.CreateImpressionEvent(config, experiment, variation.Id, userId, userAttributes);
+                var userEvent = UserEventFactory.CreateImpressionEvent(config, experiment, variation.Id, userId, userAttributes);
+                EventProcessor.Process(userEvent);
                 Logger.Log(LogLevel.INFO, string.Format("Activating user {0} in experiment {1}.", userId, experiment.Key));
-                Logger.Log(LogLevel.DEBUG, string.Format("Dispatching impression event to URL {0} with params {1}.",
-                    impressionEvent.Url, impressionEvent.GetParamsAsJson()));
 
-                try
+                // Kept For backwards compatibility.
+                // This notification is deprecated and the new DecisionNotifications
+                // are sent via their respective method calls.
+                if (NotificationCenter.GetNotificationCount(NotificationCenter.NotificationType.Activate) > 0)
```

Comment:
This check does not look great. Should we not try to call this method and this becomes a no-op if there are no listeners like earlier?

Reply:
I really like your suggestion but Activate and Track are already obsoleted, this condition will be removed in the upcoming versions and also we are overriding delegates, doesn't have the same pattern we have in java.